### DISCUSSION TITLE
Develop use cases of ENS by Darwinia Msgport

### DISCUSSION THREAD

**Discussion by hackfisher** [darwinia-network/darwinia#1380] [Ideas]
In the [recent tweet](https://x.com/VitalikButerin/status/1742540651797823948?s=20) of vitalik, he mention the importance of integration between ENS and cross-chain message services like CCIP, as Darwinia Msgport can provide more message lines and better developer experience  than CCIP. 

With vitalik's call, application development in ENS-related fields will accelerate. We should join this market and develop related use cases. cc @hujw77 @wuminzhe

**Comment by hackfisher** [comment]
Related using plans of DCDAO: https://github.com/orgs/dcdao/discussions/66

**Comment by hackfisher** [comment]
https://clusters.xyz/

**Reply by hackfisher** [reply]
clusters seems not clear and over complex, as I understand, there is cluster can do but ENS can not do.

**Comment by nblogist** [comment]
.@hackfisher do you have a rough architecture in your mind that you'd prefer? I am trying to have an internal conversation about it rn

**Reply by hackfisher** [reply]
The solution is from ENS official documents, we only need to follow it:
https://docs.ens.domains/resolvers/ccip-read

https://github.com/ringecosystem/collaboration/issues/61

**Comment by hackfisher** [comment]
Conclusion:
ENS has largely integrated CCIP-read to provide chain-agnostic solutions at the client standard level. Msgport, on the other hand, is more focused on enabling smart contracts to handle cross-chain messages and transactions. At present, Msgport is not essential for addressing ENS-related issues.
